guava/src/com/google/common/base/Ascii.java

```diff
@@ -629,4 +630,78 @@ private static int getAlphaIndex(char c) {
     // Fold upper-case ASCII to lower-case and make zero-indexed and unsigned (by casting to char).
     return (char) ((c | 0x20) - 'a');
   }
+
+  /**
+   * Returns the index within {@code sequence} of the first occurrence of the
+   * specified {@code subSequence}, ignoring the case of any ASCII alphabetic characters
+   * between {@code 'a'} and {@code 'z'} or {@code 'A'} and {@code 'Z'} inclusive, or
+   * {@code -1}, if no occurence is found.
+   *
+   * @since 16.0 // TODO: which number?
+   */
+  public static int indexOfIgnoreCase(CharSequence sequence, CharSequence subSequence) {
+    return toLowerCase(sequence).indexOf(toLowerCase(subSequence));
+  }
+
+  /**
+   * Indicates whether the character sequence {@code sequence} contains the {@code subSequence},
+   * ignoring the case of any ASCII alphabetic characters between {@code 'a'} and {@code 'z'}
+   * or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @since 16.0 // TODO: which number?
+   */
+  public static boolean containsIgnoreCase(CharSequence sequence, CharSequence subSequence) {
+    // Calling length() is the null pointer check (so do it before we can exit early).
+    int length = sequence.length();
+    if (sequence == subSequence) {
+      return true;
+    }
+    // if subSequence is longer than sequence, it is impossible for sequence to contain subSequence
+    if (subSequence.length() > length) {
+      return false;
+    }
+    return indexOfIgnoreCase(sequence, subSequence) > -1;
+  }
+
+  /**
+   * Returns if the character sequence {@code seq} starts with the character sequence {@code prefix}
+   * at the specified {@code offset}, ignoring the case of any ASCII alphabetic characters
+   * between {@code 'a'} and {@code 'z'} or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @since 16.0 // TODO: which number?
+   */
+  public static boolean startsWithIgnoreCase(CharSequence seq, CharSequence prefix, int offset) {
+    // an empty prefix starts everywhere in the sequence
+    if (prefix.length() == 0) {
+      return true;
+    }
+    // if the prefix is longer than the sequence itself or from the offset, it is impossible
+    // for the sequence to contain the prefix
+    if (prefix.length() > seq.length() || prefix.length() > seq.length() - offset) {
+      return false;
+    }
+    return indexOfIgnoreCase(seq, prefix) == offset;
```

Comment:
What do you expect `startsWithIgnoreCase("aaa", "a", 1)` to return? I'd vote for true, but this is impossible to implement without a `indexOfIgnoreCase(seq, prefix, offset)`.

Reply:
This one is 
1. logically flawed
2. Too slow (You should just use a for loop to check. If the length of prefix is m, then the time complexity is O(m) maximum)

Reply:
Thanks for noticing! I fixed it in the latest commit.